### PR TITLE
Implement spectral normalization

### PR DESCRIPTION
**Pull request by EsdeathYZH**
- [ ] nn.utils.spectral_norm api docs
- [ ] unit test case

api docs:

### CODE REVIEW

File: python/oneflow/nn/utils/spectral_norm.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        if isinstance(hook, SpectralNormStateDictHook) and hook.fn.name == name:
+            del module._state_dict_hooks[k]
+            break
+
+    for k, hook in module._load_state_dict_pre_hooks.items():
+        if isinstance(hook, SpectralNormLoadStateDictPreHook) and hook.fn.name == name:
+            del module._load_state_dict_pre_hooks[k]
+            break
+
+    return module
```
Comment (zhongshsh) [inline python/oneflow/nn/utils/spectral_norm.py:250]: 文件最后多打一行空行就不会有这个错误提示了

File: python/oneflow/nn/utils/spectral_norm.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+O_module = TypeVar('O_module', bound=Module)
+
+def spectral_norm(module: O_module,
+                  name: str = 'weight',
+                  n_power_iterations: int = 1,
+                  eps: float = 1e-12,
+                  dim: Optional[int] = None) -> O_module:
+    r"""Applies spectral normalization to a parameter in the given module.
+
```
Comment (zhongshsh) [inline python/oneflow/nn/utils/spectral_norm.py:179]: 虽然文档说明和 [PyTorch](https://pytorch.org/docs/stable/_modules/torch/nn/utils/spectral_norm.html#spectral_norm) 一致，但是可能是文档解析的问题，导致不是很美观，可以参考 [howto_write_api_doc](https://github.com/Oneflow-Inc/OneTeam/blob/master/tutorial/howto_write_api_docs.md) 添加参数说明。

File: python/oneflow/nn/utils/spectral_norm.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+                                   "the same parameter {}".format(name))
+
+        fn = SpectralNorm(name, n_power_iterations, dim, eps)
+        weight = module._parameters[name]
+        if weight is None:
+            raise ValueError(f'`SpectralNorm` cannot be applied as parameter `{name}` is None')
+        # if isinstance(weight, flow.nn.parameter.UninitializedParameter):
+        #     raise ValueError(
+        #         'The module passed to `SpectralNorm` can\'t have uninitialized parameters. '
+        #         'Make sure to run the dummy forward before applying spectral normalization')
```
Comment (zhongshsh) [inline python/oneflow/nn/utils/spectral_norm.py:103]: 没有用的注释可以删除的

Comment (zhongshsh) [reply python/oneflow/nn/utils/spectral_norm.py:103]: 我看 PyTorch 是保留了这部分内容的，这里为啥注释掉？

File: python/oneflow/nn/utils/spectral_norm.py (hunk 1 above)
Comment (EsdeathYZH) [reply python/oneflow/nn/utils/spectral_norm.py:328]: Done

File: python/oneflow/nn/utils/spectral_norm.py (hunk 3 above)
Comment (EsdeathYZH) [reply python/oneflow/nn/utils/spectral_norm.py]: 这里是因为没有flow.nn.parameter.UninitializedParameter这个类，参考了一下WeightNorm也是直接删掉了

Comment (EsdeathYZH) [reply python/oneflow/nn/utils/spectral_norm.py]: 已删除

File: python/oneflow/nn/utils/spectral_norm.py (hunk 2 above)
Comment (EsdeathYZH) [reply python/oneflow/nn/utils/spectral_norm.py:242]: 这块具体是哪里不美观啊，函数签名吗？

Comment (zhongshsh) [reply python/oneflow/nn/utils/spectral_norm.py:242]: image
image

参数显示很不友好:dizzy_face:，例如这里的 O_module 出现在文档里就很让人摸不到头脑
